src/layer/GeoJSON.js

```diff
@@ -265,12 +265,13 @@ export function latLngToCoords(latlng, precision) {
 // @function latLngsToCoords(latlngs: Array, levelsDeep?: Number, closed?: Boolean): Array
 // Reverse of [`coordsToLatLngs`](#geojson-coordstolatlngs)
 // `closed` determines whether the first point should be appended to the end of the array to close the feature, only used when `levelsDeep` is 0. False by default.
+// Check for flat arrays required to ensure unbalanced arrays are correctly converted in recursion

```

Comment:
Nit: I'd move the comment inside so that it's not exposed as public documentation.